lib/rules/hook-use-state.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @fileoverview Ensure symmetric naming of useState hook value and setter variables
+ * @author Duncan Beevers
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+const USE_STATE_ERROR_MESSAGE = 'useStateErrorMessage';
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Ensure symmetric naming of useState hook value and setter variables',
+      category: 'Best Practices',
+      recommended: false,
+      url: docsUrl('hook-use-state'),
+    },
+    fixable: 'code',
+    messages: {
+      [USE_STATE_ERROR_MESSAGE]: 'useState call is not destructured into value + setter pair',
+    },
+    schema: [{
+      type: 'object',
+      additionalProperties: false,
+    }],
```

Comment:
```suggestion
    schema: [],
```
otherwise passing a config of `{}` is allowed